src/python/pants/backend/build_files/fmt/ruff/register.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.backend.build_files.fmt.base import FmtBuildFilesRequest
+from pants.backend.python.lint.ruff import subsystem as ruff_subsystem
+from pants.backend.python.lint.ruff.rules import _run_ruff_fmt
+from pants.backend.python.lint.ruff.subsystem import Ruff
+from pants.backend.python.subsystems.python_tool_base import get_lockfile_interpreter_constraints
+from pants.core.goals.fmt import FmtResult
+from pants.engine.rules import collect_rules, rule
+from pants.util.logging import LogLevel
+
+
+class RuffRequest(FmtBuildFilesRequest):
+    tool_subsystem = Ruff
+
+
+@rule(desc="Format with Ruff", level=LogLevel.DEBUG)
+async def ruff_fmt(request: RuffRequest.Batch, ruff: Ruff) -> FmtResult:
+    ruff_ics = await get_lockfile_interpreter_constraints(ruff)
+    return await _run_ruff_fmt(request, ruff, ruff_ics)
+
+
+def rules():
+    return [
+        *collect_rules(),
+        *RuffRequest.rules(),
+        *ruff_subsystem.rules(),
+    ]
```

Comment:
Maybe you need to register this?
```suggestion
def rules():
    return [
        *collect_rules(),
        *RuffRequest.rules(),
        *ruff_subsystem.rules(),
        *_run_ruff_fmt.rules()
    ]
```

Reply:
mmhm, doesn't look like it

Reply:
it works if you add it in "src/python/pants/core/goals/update_build_files.py". I'm not sure why it's not getting pulled in when it's here. I'll have more time to check this weekend

Reply:
that's suspect. good find, but just to be clear; not a suggested solution, right? we don't want to have dependencies from core to backends.

Reply:
Some of that came up when we were redoing the rules to create lockfiles. The problem is that Pants understands a hierarchy where a language/ecosystem has tools in it, and those tools respond to goals. This works well for everything _except_ operating on build files. We want formatting build files to not need enabling the python backend, but that's a problem for 2 reasons:
1. the build files are Python, so there's overlap with some of those tools
2. Pants is Python, and uses the python backend to run tools (through PEXes).

2️⃣ comes up every so often when someone _isn't_ using the python backend but needs to set `[python].interpreter_constraints` (eg terraform needs to run a little helper program to do the dep inference). 
1️⃣ is a bigger problem. As you pointed out, it's suspect to (sneakily) include the python backend for getting build files to load. The solution we came to is that we can use a simplified version of the tool's rules, by forexample omitting dependency inference rules. One of the consequences of this simplified include is the `_run_ruff_fmt` as a function, not a rule. The fact that including _all_ the rules resolves this indicates the challenge in implementing a simplified backend. It can be challenging to find out which import you've missed.